yt/frontends/flash/fields.py

```diff
@@ -147,10 +147,14 @@ def _nele(field, data):
             return data["flash","dens"]*data["flash","ye"]*Na_code
         self.add_field(('flash','nele'), sampling_type="cell",  function=_nele, units="code_length**-3")
         self.add_field(('flash','edens'), sampling_type="cell",  function=_nele, units="code_length**-3")
+        self.add_field(('gas','El_number_density'), sampling_type="cell",  function=_nele,
+                       units=unit_system["number_density"])
         def _nion(field, data):
             Na_code = data.ds.quan(Na, '1/code_mass')
             return data["flash","dens"]*data["flash","sumy"]*Na_code
         self.add_field(('flash','nion'), sampling_type="cell",  function=_nion, units="code_length**-3")
+        self.add_field(('gas','MM_nuclei_density'), sampling_type="cell",  function=_nion,
```

Comment:
What does `MM` correspond to here?

Reply:
Multi temperature means different species/component in the simulations have different temperatures, e.g ion/electron/radiation temperature in a 3T hydrodynamics, or electron/radiation temperature in a two fluid code.

Reply:
Oh I see, `MM` comes from a direct reading of YTEP-0003. In that document `MM` is a placeholder for the species name. We don't have a generic `MM_nuclei_density` field in yt. Instead, you should be using a specific species name. If `nion` doesn't correspond to a specific species, then just `ion_density` is probably a sufficiently universal name. Maybe @jzuhone or @brittonsmith could weigh in?

Reply:
By MM, I mean the totality of atom/molecules (regardless of its ionization state and nuclei number Z)

Reply:
In that case I think just `ion_density` is the correct name. Like I mentioned in my comment, `MM` is not actually a shorthand we use inside yt, YTEP-0003 is just using it as a placeholder.